collections.go

```diff
@@ -1,17 +1,27 @@
 package collections
 
 import (
+	"context"
 	"errors"
+
+	"github.com/gogo/protobuf/io"
 )
 
 // ErrNotFound is returned when an object is not found.
 var ErrNotFound = errors.New("collections: not found")
 
-// Namespace defines a storage namespace which must be unique in a single module
+// Prefix defines a storage namespace which must be unique in a single module
 // for all the different storage layer types: Map, Sequence, KeySet, Item, MultiIndex, IndexedMap
-type Namespace uint8
+type Prefix uint8
+
+func (n Prefix) Prefix() []byte { return []byte{uint8(n)} }
 
-func (n Namespace) Prefix() []byte { return []byte{uint8(n)} }
+type Collection interface {
+	Descriptor() CollectionDescriptor
+	InitGenesis(context.Context, io.ReadCloser) error
+	ExportGenesis(context.Context, io.WriteCloser) error
+	Decode(key, value []byte) (k, v any, err error)
```

Comment:
wondering why these need to be untyped ? What about exposing the value encoder and key encoder?

Reply:
They need to be untyped because the schema itself is heterogeneous. It consists of many collections with different types

Reply:
For comparison this is how ORM exposes logical decoding entries: https://github.com/cosmos/cosmos-sdk/blob/main/orm/encoding/ormkv/entry.go